pkg/sync/sync_context.go

```diff
@@ -792,7 +794,10 @@ func (sc *syncContext) autoCreateNamespace(tasks syncTasks) syncTasks {
 					}
 				}
 			} else if apierr.IsNotFound(err) {
-				tasks = append(tasks, &syncTask{phase: common.SyncPhasePreSync, targetObj: unstructuredObj, liveObj: nil})
+				unstructuredObjCopy := unstructuredObj.DeepCopy()
+				if sc.namespaceCreator(unstructuredObjCopy) {
+					tasks = append(tasks, &syncTask{phase: common.SyncPhasePreSync, targetObj: unstructuredObjCopy, liveObj: nil})
+				}
```

Comment:
My expectation as a user would be that when I set the `createNamespaceMetadata` field, all auto-created namespaces will be updated with that metadata.

What if in `WithNamespaceCreation` we accepted multiple `namespaceModifier` functions and run them in series where they are currently being run?

Reply:
@crenshaw-dev  Do you want to add ns metadata to the already created auto namespace?

Reply:
yes, use case is 

User creates an app with autocreate namespaces
User syncs the app, and namespaces are auto-created
User adds a new label to the createNamespaceMetadata.labels field
User syncs the app, and the label is applied to already-auto-created namespaces


Working on this

Reply:
@ashutosh16 @crenshaw-dev this use case is covered now 

https://github.com/argoproj/argo-cd/pull/10288/files#diff-627c32a4f31d0f97e6ddc1f19ab3add7fb7791e0d1b0503d78c6c049e8cb4865R260

when the app is going to be synced namespaceModifier will be called and labels will be added to namespace

Reply:
lgtm!